MapboxNavigation/MGLMapView.swift

```diff
@@ -46,24 +46,25 @@ extension MGLMapView {
         }
     }
     
-    func addArrow(_ routeProgress: RouteProgress) {
-        let maneuverCoordinate = routeProgress.currentLegProgress.upComingStep?.maneuverLocation
-        let polylineCoordinates = routeProgress.route.coordinates
+    func addArrow(route: Route, legIndex: Int, stepIndex: Int) {
+        let step = route.legs[legIndex].steps[stepIndex]
```

Comment:
If the user manages to swipe past the last step in the leg or the last step in the route, this line will crash.